File: app/scala/src/main/scala/com/example/accounting/domain/financial/IncomeStatement.scala
package com.example.accounting.domain.financial

import java.time.LocalDate

/**
 * 損益計算書（Income Statement / P/L）
 *
 * @param fromDate 開始日
 * @param toDate 終了日
 * @param revenues 収益項目
 * @param expenses 費用項目
 * @param grossProfit 売上総利益
 * @param operatingIncome 営業利益
 * @param netIncome 当期純利益
 * @param totalRevenues 総収益
 * @param totalExpenses 総費用
 */
case class IncomeStatement(
    fromDate: LocalDate,
    toDate: LocalDate,
    revenues: List[IncomeStatementItem],
    expenses: List[IncomeStatementItem],
    grossProfit: BigDecimal,
    operatingIncome: BigDecimal,
    netIncome: BigDecimal,
    totalRevenues: BigDecimal,
    totalExpenses: BigDecimal,
)

File: app/ruby/accounting_system/app/models/income_statement.rb
# frozen_string_literal: true

# 損益計算書（Income Statement / P/L）
class IncomeStatement
  attr_accessor :from_date, :to_date, :revenues, :expenses,
                :gross_profit, :operating_income, :net_income,
                :total_revenues, :total_expenses

  def initialize(
    from_date:,
    to_date:,
    revenues: [],
    expenses: [],
    gross_profit: 0,
    operating_income: 0,
    net_income: 0,
    total_revenues: 0,
    total_expenses: 0
  )
    @from_date = from_date
    @to_date = to_date
    @revenues = revenues
    @expenses = expenses
    @gross_profit = gross_profit
    @operating_income = operating_income
    @net_income = net_income
    @total_revenues = total_revenues
    @total_expenses = total_expenses
  end
end
